src/test/java/com/shibumi/MaximumPathSumTest.java

```diff
@@ -46,4 +62,102 @@ public void testCalculateMaxSum_secondScenario() {
         assertEquals(12, maxValue);
     }
 
+
+    /**
+     * Input grid:
+     *
+     * 1 3 -1
+     * -1 -5 1
+     * 4 2 1
+     *
+     **/
```

Comment:
What if all the values are negative? I think there's a bug in the algorithm that will cause it to fail in that case.

Reply:
great catch, I just refactored the code and added a test case for that